### ISSUE TITLE
TypeError: cb is not a function

### ISSUE THREAD

**Issue by efkan** [jdesboeufs/connect-mongo#272]
I've encountered this error. When logging out from the web application it throws a `TypeError`.

My code that is in use:

```
var mongoose = require('mongoose')
mongoose.Promise = global.Promise 
var session = require('express-session')
var MongoStore = require('connect-mongo')(session)

app.use(session({
  secret: "verySecretKey",
  resave: true,
  saveUninitialized: true,
  cookie: { maxAge: 19 * 60000 }, // store for 19 minutes
  store: new MongoStore({
	mongooseConnection: mongoose.connection
  })
}))

process.on('unhandledRejection', (reason, p) => {
  console.log('app.js - Unhandled Rejection at: \nPromise', p, '\nreason:', reason);
});
```

The code which runs when logging out:

```
router.get(['/login','/logout'], function(req, res){
  req.session.destroy()
  res.redirect("/")
})
```

Console output after logout:

```
app.js - Unhandled Rejection at: 
 Promise Promise {
  <rejected> TypeError: cb is not a function
    at promise.then.res (/home/efkan/Projects/awesome/node_modules/connect-mongo/src/index.js:7:18)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7) } 
reason: TypeError: cb is not a function
    at promise.then.res (/home/efkan/Projects/awesome/node_modules/connect-mongo/src/index.js:7:18)
    at <anonymous>
    at process._tickDomainCallback (internal/process/next_tick.js:228:7)
```

The error specifies [this line](https://github.com/jdesboeufs/connect-mongo/blob/master/src/index.js#L7) of `connect-mongo` source code

**Development environment:**

Module Name | Version |
-|-|
connect-mongo | 2.0.0
mongoose | 4.12.4
express-session | 1.9.3
express | 4.10.6

**Comment by lucduong** [comment]
I got the same error.

**Comment by SamJWeissman** [comment]
@efkan @lucduong 

Any ideas on this one? Seeing the same thing.

**Comment by lucduong** [comment]
@SamJWeissman I just downgrade the version to `1.3.2` and everything works well.

```yarn add connect-mongo@1.3.2```

**Comment by SamJWeissman** [comment]
@lucduong 

Awesome, thanks!

**Comment by jdesboeufs** [comment]
It’s a small fix.
I will look at this ASAP.

**Comment by jdesboeufs** [comment]
You have to pass a callback in your `destroy` call.
`req.session.destroy(cb)`

**Comment by brainthinks** [comment]
I am having this problem too.  From what I see, it is (potentially) an oversight in the `withCallback` function defined at the top of `src/index.js`.  In my project, I am using promises rather than callbacks, therefore, when I call the public `get` API method, I am not sending in a callback.  It shouldn't be needed if using promises or async/await.  However, the `get` method uses `withCallback`, which REQUIRES cb to be defined - it does not perform a null check on cb.

@jdesboeufs - unless I'm missing something, performing a null check on cb to make it not required to allow promises only would fix this issue.  

If you agree, I can create a pull request.

**Comment by jdesboeufs** [comment]
@brainthinks I agree!

**Comment by brainthinks** [comment]
@jdesboeufs - cool.  Just added #276

**Comment by jdesboeufs** [comment]
Fixed in version 2.0.1.
Thx @brainthinks!!
